addons/loyalty/models/loyalty_program.py

```diff
@@ -184,7 +184,7 @@ def _compute_coupon_count(self):
     def _compute_is_nominative(self):
         for program in self:
             program.is_nominative = program.applies_on == 'both' or\
-                (program.program_type == 'ewallet' and program.applies_on == 'future')
+                (program.program_type in ('ewallet', 'loyalty') and program.applies_on == 'future')
```

Comment:
I'm not sure I understand what this changes and why it's necessary.
What's exactly the impact/use of `is_nominative` ?

Reply:
`is_nominative` is used to distinguish between programs that are only available to specific partner and public programs that can be used for any partners :thinking:

https://github.com/odoo/odoo/pull/133755#discussion_r1346900284

See this point mentioned in the task:  
```
Partner is not linked for newly created loyalty card for the Use point on "Future Order". Ref-https://youtu.be/89P3A6wMgMk  (For "Current & Future orders" its working fine)

DALA: This as to be changed but only for Loyalty cards. "Next order coupons" should not be linked to a partner.. 
```